src/render/painter.js

```diff
@@ -296,6 +295,35 @@ class Painter {
         return new StencilMode({func: gl.EQUAL, mask: 0xFF}, this._tileClippingMaskIDs[tileID.key], 0x00, gl.KEEP, gl.KEEP, gl.REPLACE);
     }
 
+    /*
+     * Sort coordinates by Z as drawing tiles is done in Z-descending order.
+     * All children with the same Z write the same stencil value.  Children
+     * stencil values are greater than parent's.  Stencil ref values continue
+     * range used in _tileClippingMaskIDs.
```

Comment:
Good strategy; clarifying question: This is only supposed to work for raster + DEM tiles where the geometry doesn't protrude from the tile's viewport, correct?

Reply:
Clarified it in updated comment.